src/components/shared/ManagedTree.js

```diff
@@ -40,21 +40,18 @@ type State = {
 
 class ManagedTree extends Component<Props, State> {
   constructor(props: Props) {
-    super();
-
+    super(props);
     this.state = {
       expanded: props.expanded || new Set(),
       focusedItem: null
     };
   }
 
-  componentWillReceiveProps(nextProps: Props) {
-    const listItems = nextProps.listItems;
+  componentWillReceiveProps({ listItems, highlightItems, focused }: Props) {
     if (listItems && listItems != this.props.listItems && listItems.length) {
```

Comment:
minor nit:

```js
   componentWillReceiveProps(nextProps: Props) {
     const {listItems} = this.props;
```

typically we're destructuring on the next line and using props

Reply:
I destructed the nextProps because I noticed we only used those specific props.